src/plugins/vis_type_vega/public/vega_view/vega_map_view/layers/vega_layer.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import type { Map, CustomLayerInterface } from 'mapbox-gl';
+import type { LayerParameters } from './types';
+
+// @ts-ignore
+import { vega } from '../../lib/vega';
+
+export interface VegaLayerContext {
+  vegaView: vega.View;
+  updateVegaView: (map: Map, view: vega.View) => void;
+}
+
+export function initVegaLayer({
+  id,
+  map: mapInstance,
+  context: { vegaView, updateVegaView },
+}: LayerParameters<VegaLayerContext>) {
+  const vegaLayer: CustomLayerInterface = {
```

Comment:
this is implemented as two overlaying divs. sorry, I missed this at first glance when we talked in person, thinking you user the mb-customlayerinterface to render into the context.

This is similar to how the leaflet-implementation worked, also good :)